### DISCUSSION TITLE
Allow more file names and ".yml" extension

### DISCUSSION THREAD

**Discussion by ToshY** [louislam/dockge#10] [Feature Request]
**Problem**

Mounted existing stacks currently have some other (older) variants of compose file names and extenions, e.g. `docker-compose.yaml`, `docker-compose.yml`, or `compose.yml`.

```text
collection/
├── dockge/
│   └── docker-compose.yml
├── kuma/
│   └── compose.yml
└── registry/
    └── docker-compose.yaml
```

Using `compose.yml` for `kuma`, Dockge will not recognise the config and containers in the UI.

![image](https://github.com/louislam/dockge/assets/31921460/4a21b20b-d601-48c2-abc3-60c276fd664a)

Same goes for other stacks that use `docker-compose.yml` or `docker-compose.yaml`.

**Propose**

Allow multiple compose file names and `yml` extenion.

* `docker-compose.yml`
* `docker-compose.yaml`
* `compose.yml`

---

> **Sidenote**
It also currently shows directories in dockge that do not contain any compose configurations.
![image](https://github.com/louislam/dockge/assets/31921460/24f17b3f-2089-43fd-b10a-ce88be123f25)
Think dockge should identify if a directory contains a compose configuration, and if not, do not show it in UI; related to https://github.com/louislam/dockge/discussions/9

**Comment by Zaid-maker** [comment]
Good Catch!

**Comment by CobreDev** [comment]
For anyone else having this issue, I was able to get around it this way. 
I had everything labelled "compose.yml", but "compose.y***a***ml" is what dockge is looking for. Using this command I was able to recursively rename all my compose files to end in `.yaml` and they're now working in dockge:
`autoload zmv`
`zmv -n '(**/)(*)compose.yml' '$1compose.yaml'`
note: `-n` will do a dry run and list the files to be moved, so remove this once you've confirmed that it's going to move the correct files

This will find any file ending in `compose.yml` (this will include files named `docker-compose.yml`) and rename them to `compose.yaml`.

Use at your own risk, but this worked for me!

**Reply by theriaultjulien** [reply]
Your command didn't work for me on Ubuntu 22.04, I used this one instead.
`find /path/to/your-compose-files/ -type f -name 'docker-compose.yml' -exec rename 's/docker-compose\.yml/compose.yaml/' {} \;`

This will rename all files named `docker-compose.yml` in the specified directory and all subdirectories to `compose.yaml`.

You might need to install `rename` package using `apt install rename` before running this command.

**Comment by louislam** [comment]
1.1.0 is released with this feature!

**Comment by almas-x** [comment]
Is the format /opt/stacks/<stackName>/<stackName>.yaml supported?
